app/src/main/java/net/squanchy/tweets/service/TwitterRepository.java

```diff
@@ -0,0 +1,53 @@
+package net.squanchy.tweets.service;
+
+import com.twitter.sdk.android.core.Callback;
+import com.twitter.sdk.android.core.Result;
+import com.twitter.sdk.android.core.TwitterCore;
+import com.twitter.sdk.android.core.TwitterException;
+import com.twitter.sdk.android.core.models.Search;
+import com.twitter.sdk.android.core.services.SearchService;
+
+import io.reactivex.Observable;
+import io.reactivex.ObservableEmitter;
+import retrofit2.Call;
+
+public class TwitterRepository {
+
+    private static final int MAX_ITEM_PER_REQUEST = 100;
+
+    private final SearchService searchService;
+    private final String query;
+
+    public TwitterRepository(String query) {
```

Comment:
Can we pass in the query to `load()` instead? So we don't have to pass it aaaall the way through Dagger which is a bit weird